File: expo-maps/android/src/main/java/expo/modules/maps/GoogleMapsGestures.kt
package expo.modules.maps

import com.google.android.gms.maps.GoogleMap

class GoogleMapsGestures(private val googleMap: GoogleMap) : Gestures {

  override fun setEnabledRotateGesture(enabled: Boolean) {
    googleMap.uiSettings.isRotateGesturesEnabled = enabled
  }

  override fun setEnabledScrollGesture(enabled: Boolean) {
    googleMap.uiSettings.isScrollGesturesEnabled = enabled
  }

  override fun setEnabledTiltGesture(enabled: Boolean) {
    googleMap.uiSettings.isTiltGesturesEnabled = enabled
  }

  override fun setEnabledZoomGesture(enabled: Boolean) {
    googleMap.uiSettings.isZoomGesturesEnabled = enabled
  }

  override fun setEnabledAllGestures(enabled: Boolean) {
    googleMap.uiSettings.setAllGesturesEnabled(enabled)
  }
}

File: expo-maps/ios/ExpoMaps/GoogleMaps/GoogleMapsGestures.swift
import GoogleMaps

final class GoogleMapsGestures: Gestures {

  internal var mapView: GMSMapView

  init(mapView: GMSMapView) {
    self.mapView = mapView
  }

  init?(coder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  func setEnabledRotateGesture(enabled: Bool) {
    mapView.settings.rotateGestures = enabled
  }

  func setEnabledScrollGesture(enabled: Bool) {
    mapView.settings.scrollGestures = enabled
  }

  func setEnabledTiltGesture(enabled: Bool) {
    mapView.settings.tiltGestures = enabled
  }

  func setEnabledZoomGesture(enabled: Bool) {
    mapView.settings.zoomGestures = enabled
  }

  func setEnabledAllGestures(enabled: Bool) {
    mapView.settings.setAllGesturesEnabled(enabled)
  }
}
